core/async/src/instrumentation/queue.rs

```diff
@@ -0,0 +1,60 @@
+use core::str;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::atomic::AtomicU64;
+
+use parking_lot::RwLock;
+use serde::Serialize;
+
+pub struct InstrumentedQueue {
+    pending: RwLock<HashMap<String, AtomicU64>>,
+}
+
+impl InstrumentedQueue {
+    pub fn new(_queue_name: &str) -> Arc<Self> {
```

Comment:
nit: Can we get rid of this?

Reply:
You mean `_queue_name`? it will be used for metrics.